app/models/vm_or_template/operations.rb

```diff
@@ -93,6 +95,25 @@ def validate_vm_control
     nil
   end
 
+  included do
+    supports :control do
+      msg = if vm.retired?
+              'The VM is retired'
```

Comment:
Probably just `N_()` and have the caller call `_()` if needed. But seems we need these strings in the catalog.